.github/workflows/aws-s3-bundle-update.yaml

```diff
@@ -0,0 +1,82 @@
+name: AWS-S3 DataConnector Bundle Auto-Update
+run-name: Auto-updating AWS-S3 bundles for ${{ github.event.pull_request.head.ref }}
+
+on:
+  pull_request:
+    branches:
+      - master
+    paths:
+      # Trigger when any of these files in AWS-S3 directory change
+      - 'DataConnectors/AWS-S3/*.ps1'
+      - 'DataConnectors/AWS-S3/*.py'
+      - 'DataConnectors/AWS-S3/*.md'
+      - 'DataConnectors/AWS-S3/CloudFormation/**'
+      - 'DataConnectors/AWS-S3/Enviornment/**'
+      - 'DataConnectors/AWS-S3/Utils/**'
+      # Don't trigger on zip file changes (to avoid recursion)
+      - '!DataConnectors/AWS-S3/*.zip'
+  
+  # Allow manual workflow dispatch for testing
+  workflow_dispatch:
+
+jobs:
+  auto-update-bundles:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+      pull-requests: write
+    
+    steps:
+      - name: Checkout PR branch with sparse checkout
+        uses: actions/checkout@v4
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
```

Comment:
The workflow checks out the PR branch and executes the bundling script from that branch with contents: write permission. A malicious PR can modify bundleAwsS3Scripts.sh to execute arbitrary commands and push malicious changes back to the repository. You can restore/checkout the original script from master branch before executing it.
I would also recommend getting security review done for this workflow.

Reply:
@rahul0216 

This repo has several workflows that are committing changes to PR branches similar to this one, I'll check to how those are doing so in a secure manner.

Who is the contact for a security review for this repo?